homeassistant/components/broadlink/climate.py

```diff
@@ -0,0 +1,250 @@
+"""
+Support for Chinese wifi thermostats (Floureon, Beok, Beca Energy).
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/climate.broadlink/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.climate import PLATFORM_SCHEMA, ClimateDevice
+from homeassistant.components.climate.const import (
+    DOMAIN, STATE_AUTO, STATE_HEAT, SUPPORT_AWAY_MODE, SUPPORT_OPERATION_MODE,
+    SUPPORT_TARGET_TEMPERATURE)
+from homeassistant.const import (
+    ATTR_ENTITY_ID, ATTR_TEMPERATURE, CONF_FRIENDLY_NAME, CONF_HOST, CONF_MAC,
+    PRECISION_HALVES, STATE_OFF, TEMP_CELSIUS)
+import homeassistant.helpers.config_validation as cv
+
+
+REQUIREMENTS = ['BroadlinkWifiThermostat==2.4.1']
+
+CONF_EXTERNAL_TEMP = 'external_temp'
+CONF_AWAY_TEMP = 'away_temp'
+
+ATTR_LOOP_MODE = 'loop_mode'
+ATTR_SEN = 'sen'
+ATTR_OSV = 'osv'
+ATTR_DIF = 'dif'
+ATTR_SVH = 'svh'
+ATTR_SVL = 'svl'
+ATTR_ADJ = 'adj'
+ATTR_FRE = 'freeze'
+ATTR_PON = 'pon'
+
+ATTR_WEEK_START_1 = 'week_start_1'
+ATTR_WEEK_STOP_1 = 'week_stop_1'
+ATTR_WEEK_START_2 = 'week_start_2'
+ATTR_WEEK_STOP_2 = 'week_stop_2'
+ATTR_WEEK_START_3 = 'week_start_3'
+ATTR_WEEK_STOP_3 = 'week_stop_3'
+
+ATTR_WEEKEND_START = 'weekend_start'
+ATTR_WEEKEND_STOP = 'weekend_stop'
+ATTR_AWAY_TEMP = 'away_temp'
+ATTR_HOME_TEMP = 'home_temp'
+
+SERVICE_SET_SCHEDULE = 'broadlink_set_schedule'
+SERVICE_SET_ADVANCED_CONF = 'broadlink_set_advanced_conf'
+
+# Validation of the user's configuration
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_MAC): cv.string,
+    vol.Required(CONF_FRIENDLY_NAME): cv.string,
```

Comment:
The friendly name is required? That is not usual...
I see the library requires it... but it isn't used actually?

![image](https://user-images.githubusercontent.com/195327/54224658-3d272880-44fa-11e9-94d3-147d0dd4bad3.png)

In the above image, it shows the only part of the code that uses the `entity_id`?
The variables seem to be unused, however, they are required?